### PR TITLE
feat: Created cloud file store service, backblaze

### PR DESCRIPTION
**Pull request by TheNerdMan**
Backblaze is now working in the nest branch, when testing I'd suggest creating a free backblaze account and setting up a test bucket (you get 10GB free).

Make sure you are setting these new enviroment variables:
- MOM_STORAGE_USE_LOCAL
- MOM_STORAGE_REGION
- MOM_STORAGE_ENDPOINT_URL
- MOM_STORAGE_BUCKET_NAME
- MOM_STORAGE_ACCESS_KEY_ID
- MOM_STORAGE_SECRET_ACCESS_KEY

### CODE REVIEW

**Review by Gocnak — CHANGES_REQUESTED** [review]
We are foregoing local store in favor of the minio dev instance. You can remove the local.service.ts and the serving of assets from the backend (nest-cli.json) change

**Comment by TheNerdMan** [comment]
> the serving of assets from the backend (nest-cli.json) change

This change was mainly to have the MMod logo in properly served in the OpenAPI UI and if you use a browser to get a JWT token

File: server/src/modules/maps/maps.service.ts [hunk 1]
```diff
@@ -193,10 +188,6 @@ export class MapsService {
 
     private storeMapFile(mapFileBuffer, mapModel) {
         const fileName = `maps/${mapModel.name}.bsp`;
-        if (appConfig.storage.useLocal) {
```
Comment (Gocnak) [inline server/src/modules/maps/maps.service.ts]: And remove `useLocal` from the config.

**Review by Gocnak — CHANGES_REQUESTED** [review]

File: server/src/modules/maps/maps.service.ts (hunk 1 above)
Comment (TheNerdMan) [reply server/src/modules/maps/maps.service.ts]: [fixed](https://github.com/momentum-mod/website/pull/624/commits/9fb93c86be2d1183365dec5fe1193509989fd89f)

File: server/src/@common/dto/map/map.dto.ts [hunk 2]
```diff
@@ -37,7 +38,9 @@ export class MapDto implements MapDB {
         let images = _images;
         if (images == null || images.length == 0) {
             // if null then try get it from map object
-            images = (_map as any).mapimages;
+            console.log((_map as any).mapimages);
+
```
Comment (Gocnak) [inline server/src/@common/dto/map/map.dto.ts:42]: Leftover console.log ?

File: server/src/@common/filestore/cloud.service.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { appConfig } from 'config/config';
+import { FileStoreUtilsService } from './utils.service';
+import { IFileStoreCloudFile } from './fileStore.interface';
+import { S3Client, PutObjectCommand, DeleteObjectCommand } from '@aws-sdk/client-s3';
+
+let s3Client: S3Client;
```
Comment (Gocnak) [inline server/src/@common/filestore/cloud.service.ts:7]: Does this need to be outside the class or can it be in the FileStoreCloudService constructor as `this.s3Client = new S3Client(...` ?

**Review by Gocnak — CHANGES_REQUESTED** [review]

File: server/src/modules/maps/maps.controller.ts [hunk 4]
```diff
@@ -46,7 +47,7 @@ export class MapsController {
         return this.mapsService.Get(mapID);
     }
 
-    @Post()
+    @Post('')
```
Comment (tsa96) [inline server/src/modules/maps/maps.controller.ts:50]: What's the need for this, doesn't Nest treat these two identically?

**Review by tsa96 — CHANGES_REQUESTED** [review]
Awesome work! S3 stuff is running fine for me.

File: server/src/@common/dto/map/map.dto.ts (hunk 2 above)
Comment (TheNerdMan) [reply server/src/@common/dto/map/map.dto.ts:42]: Yes will remove

File: server/src/modules/maps/maps.controller.ts (hunk 4 above)
Comment (TheNerdMan) [reply server/src/modules/maps/maps.controller.ts:50]: Yes it does. I don't remember why I did that, does it need changing?

File: server/src/@common/filestore/cloud.service.ts (hunk 3 above)
Comment (TheNerdMan) [reply server/src/@common/filestore/cloud.service.ts]: Yeah good catch, fixed

File: server/src/modules/maps/maps.controller.ts (hunk 4 above)
Comment (TheNerdMan) [reply server/src/modules/maps/maps.controller.ts]: Doesn't matter, changed now anyway

**Review by Gocnak — APPROVED** [review]

**Review by tsa96 — APPROVED** [review]

**Pull request merged**